qa-tests-backend/src/test/groovy/IntegrationsTest.groovy

```diff
@@ -519,6 +520,33 @@ class IntegrationsTest extends BaseSpecification {
                 Env.mustGetGCPAccessKey()
     }
 
+    @Unroll
+    @Tag("Integration")
+    def "Verify S3 Compatible Integration: #integrationName"() {
+        when:
+        "the integration is tested"
+        def backup = ExternalBackupService.getS3CompatibleIntegrationConfig(integrationName, endpoint, urlStyle)
+
+        then:
+        "verify test integration"
+        // Test integration for S3 compatible performs test backup (and rollback).
+        ExternalBackupService.getExternalBackupClient().testExternalBackup(backup)
```

Comment:
As far as I understand, this calls [this function](https://github.com/stackrox/stackrox/blob/b614826867f333c00141f9cbd69f7381b84d5493/central/externalbackups/service/service_impl.go#L123) that [Test](https://github.com/stackrox/stackrox/blob/26285ab0aaf817510e9f5d4b6c9b86402c76bf36/central/externalbackups/plugins/s3compatible/s3compatible.go#L236) the backup with a dummy content and then deletes it but there isn't any assertion for the actual [Backup](https://github.com/stackrox/stackrox/blob/26285ab0aaf817510e9f5d4b6c9b86402c76bf36/central/externalbackups/plugins/s3compatible/s3compatible.go#L152) being triggered by the scheduler nor rollbacks being done in the actual backups. Do you think it's worth it investing in this kind of assertion or do you think it will be extremely flaky?

Reply:
hmm good point. I don't think such tests would be flaky necessarily. But I think they would increase the test complexity quite a bit since we would likely need to add some AWS/GCP SDK code to peak into the buckets from the groovy tests. So tbh I'm not sure it's worth doing that, most problems are with integration that should be detected by the `Test` function.

Reply:
Fair. I think it's ok to merge it without it. At least for now. We can open another PR later to do the other type of test.